### PR TITLE
esp32/boards: Split out CPU frequency. 160MHz D2WD

### PR DESCRIPTION
**Pull request by nevercast**
Related https://github.com/micropython/micropython/issues/5169

Remove the 240MHz CPU config option from `sdkconfig.base` and create new `sdkconfig.` files for 160MHz and 240MHz. In summary:

* **GENERIC** is 160MHz
* **GENERIC_SPIRAM** is 160MHz 
* **GENERIC_D2WD** is 160MHz
* **TINYPICO** is 240MHz

I have not created an explicit configuration for `ESP32-S0WD`. This could be a separate pull request or I could add a new board config here. I'm just not familiar enough with the single core to immediately create that, I would need to do a little research. I also don't own an S0WD to test with and would prefer to not make the claim that the board config works if I can't test it.

Assuming we concatenate SDKCONFIG in order, `CONFIG_ESP32_DEFAULT_CPU_FREQ_160=y` could be a default in `sdkconfig.base` and then adding `sdkconfig.240mhz` to the SDKCONFIG var after `sdkconfig.base` would override the option. (I make the assumption that duplicate KConfig options override the previous one).

### CODE REVIEW

**Comment by nevercast** [comment]
@zrecore Your FemtoBeacons are D2WDs, correct? Have you had any clock related issues?

**Comment by dpgeorge** [comment]
Thanks, it looks pretty good to me.  I assume it works with IDF v3 and v4?

> Assuming we concatenate SDKCONFIG in order, `CONFIG_ESP32_DEFAULT_CPU_FREQ_160=y` could be a default in `sdkconfig.base` and then adding `sdkconfig.240mhz` to the SDKCONFIG var after `sdkconfig.base` would override the option. (I make the assumption that duplicate KConfig options override the previous one).

That could be an option, although the default could also be 80MHz.

Do you know, if the config doesn't select a clock speed, is there a default set by the IDF?  Or does it give an error if no freq is set?

File: ports/esp32/boards/TINYPICO/mpconfigboard.mk [hunk 1]
```diff
@@ -2,4 +2,5 @@ FLASH_FREQ = 80m
 
 SDKCONFIG += boards/sdkconfig.base
 SDKCONFIG += boards/sdkconfig.spiram
+SDKCONFIG += boards/sdkconfig.240mhz
```
Comment (dpgeorge) [inline ports/esp32/boards/TINYPICO/mpconfigboard.mk]: To be consistent, the 240mhz should come before spiram.

Comment (nevercast) [reply ports/esp32/boards/TINYPICO/mpconfigboard.mk]: Correct. Done.

**Comment by nevercast** [comment]
> Thanks, it looks pretty good to me. I assume it works with IDF v3 and v4? 

I cooked my last board yesterday so would like some others to test. I'll reach out on Slack. The CI all passes.

> That could be an option, although the default could also be 80MHz.

Default for `.base` and 160MHz for every other board?

> Do you know, if the config doesn't select a clock speed, is there a default set by the IDF? Or does it give an error if no freq is set?

I can check this.

**Comment by nevercast** [comment]
@dpgeorge 
image

160MHz is the IDF default.

**Comment by dpgeorge** [comment]
> 160MHz is the IDF default.

Yes indeed it is.  I tested by commenting out the `CONFIG_ESP32_DEFAULT_CPU_FREQ_240=y` line in the current sdkconfig.base and it builds fine and boots at 160MHz by default.

So I think with this PR you can just remove the sdkconfig.160MHz file entirely, and it should all work.

**Comment by dpgeorge** [comment]
> So I think with this PR you can just remove the sdkconfig.160MHz file entirely, and it should all work.

@nevercast if you agree with this I'm happy to make that change during merge of this PR.

**Comment by nevercast** [comment]
Sorry, I've been slammed at work the last couple weeks and I've accumulated a MicroPython backlog. I've rebased and pushed the changes we agreed. Should be good to merge.

**Comment by nevercast** [comment]
In summary, here are the new clock speeds:


* **GENERIC** is 160MHz
* **GENERIC_SPIRAM** is 160MHz 
* **GENERIC_D2WD** is 160MHz
* **TINYPICO** is 240MHz


I've left all the GENERIC boards as default 160MHz as I felt it was safer to run slower by default and support more boards.

**Comment by dpgeorge** [comment]
> I've left all the GENERIC boards as default 160MHz

Although this will be a user-facing change (changed from 240MHz on the provided firmware at https://micropython.org/download) I agree with it, since 160MHz is the default in the IDF.  `machine.freq(240000000)` works and can be used to increase frequency by the user.

Merged in 59c1b7d4b8fb1d441f9322976582c8709f01dfbe

**Comment by zrecore** [comment]
I haven't.

**Pull request closed without merging**